messages/src/lib.rs

```diff
@@ -299,31 +391,25 @@ pub struct Interval {
 
 impl Interval {
     pub const EMPTY: Self = Self {
-        start: Time::from_seconds_since_epoch(0),
+        start: Time::from_time_precision_units(0),
         duration: Duration::ZERO,
     };
 
-    /// Create a new [`Interval`] from the provided start and time precision.
+    /// Create a new [`Interval`] from the provided start with a duration of 1 time_precision unit.
     /// Returns an error if the end of the interval cannot be represented as a [`Time`].
     ///
-    /// This is the preferred constructor for intervals based on task time precision.
-    /// For intervals with arbitrary durations, use [`Interval::new_with_duration`].
-    pub fn new(start: Time, time_precision: TimePrecision) -> Result<Self, Error> {
-        let duration = Duration::from_seconds(time_precision.as_seconds());
-        start
-            .0
-            .checked_add(time_precision.as_seconds())
-            .ok_or(Error::IllegalTimeArithmetic("duration overflows time"))?;
-
-        Ok(Self { start, duration })
+    /// This is the preferred constructor for batch bucket intervals.
+    /// For intervals with arbitrary durations, use [`Interval::new`].
+    pub fn single(start: Time) -> Result<Self, Error> {
```

Comment:
I don't love this method name because my first thought is that it refers to the fact that the method returns a single interval and not a list, but so does `::new`. How about `minimal`, which I think better describes what sets this apart from other constructors?